src/wasm_worker.js

```diff
@@ -28,52 +27,44 @@ if (ENVIRONMENT_IS_NODE) {
     return f;
   }
 
-  var fs = require('fs');
-  var vm = require('vm');
-
-  Object.assign(global, {
+  Object.assign(globalThis, {
     self: global,
-    require,
-#if !EXPORT_ES6
-    // `vm.runInThisContext` global scope lacks `__filename` and `__dirname`
-    __filename,
-    __dirname,
-#endif
-    Worker: nodeWorkerThreads.Worker,
-    importScripts: (f) => vm.runInThisContext(fs.readFileSync(f, 'utf8'), {filename: f}),
     postMessage: (msg) => parentPort.postMessage(msg),
-    performance: global.performance || { now: Date.now },
     addEventListener: (name, handler) => parentPort.on(name, wrapMsgHandler(handler)),
     removeEventListener: (name, handler) => parentPort.off(name, wrapMsgHandler(handler)),
   });
 }
 #endif // ENVIRONMENT_MAY_BE_NODE
 
-{{{ implicitSelf() }}}onmessage = function(d) {
+var wasmWorkerArgs;
+
+onmessage = function(d) {
   // The first message sent to the Worker is always the bootstrap message.
   // Drop this message listener, it served its purpose of bootstrapping
   // the Wasm Module load, and is no longer needed. Let user code register
   // any desired message handlers from now on.
-  {{{ implicitSelf() }}}onmessage = null;
-  d = d.data;
-#if !MODULARIZE
-  self.{{{ EXPORT_NAME }}} = d;
-#endif
-#if !MINIMAL_RUNTIME
-  d['instantiateWasm'] = (info, receiveInstance) => { var instance = new WebAssembly.Instance(d['wasm'], info); return receiveInstance(instance, d['wasm']); }
+  /** @suppress {checkTypes} */
+  onmessage = null;
+#if RUNTIME_DEBUG
+  dbg('wasm worker initial onmessage');
 #endif
-#if TRUSTED_TYPES
-  // Use Trusted Types compatible wrappers.
-  if (typeof trustedTypes != 'undefined' && trustedTypes.createPolicy) {
-    var p = trustedTypes.createPolicy('emscripten#scriptPolicy1', { createScriptURL: (ignored) => d.js });
-    importScripts(p.createScriptURL('ignored'));
-  } else
-#endif
-  importScripts(d.js);
+  d = d.data;
 #if MODULARIZE
   {{{ EXPORT_NAME }}}(d);
+#else
+  wasmWorkerArgs = d;
+  wasmMemory = d['mem'];
+  updateMemoryViews();
+#endif
+#if MINIMAL_RUNTIME
+  Module['wasm'] = d['wasm'];
+  loadModule()
+#else
+  wasmModuleReceived(d['wasm']);
 #endif
   // Drop now unneeded references to from the Module object in this Worker,
   // these are not needed anymore.
-  d.wasm = d.mem = d.js = 0;
+  d['wasm'] = d['mem'] = 0;
```

Comment:
Are these quotings bugfixes, or necessary only now for some reason?

Reply:
They was not needed before since `wasm_worker.js` was generated as a separate file that was not run through closure.

Now this code is part of the main output which closure sees so quotes are needed both on the sending and receiving size.

Once we do the followup of also inlining the AUDIO_WORKLET file I think we can completly remove these quotes and let closure minify these names (like we did in the pthread case) which will further reduce code size.